src/databases/Curve3D/avtCurve3DFileFormat.C

```diff
@@ -0,0 +1,1122 @@
+        j = 0 ;
+        for (size_t i=0; i<curves.size(); i++)
+        {
+            vtkUnstructuredGrid *rg = curves[i] ;
+            vtkPoints *lpts = rg->GetPoints() ;
+            int npl = lpts->GetNumberOfPoints();
+            for (int k=1; k<npl; k++)
+            {
+                verts[0] = j + k-1;
+                verts[1] = j + k;
+                grid->InsertNextCell(VTK_LINE, 2, verts);
+            }
+            j = j + npl ;
+        }
+        return(grid) ;
+    }
+    for (size_t i = 0 ; i < curves.size() ; i++)
+    {
+        if (strcmp(("cur_" + curveNames[i]).c_str(), name) == 0)
+        {
+            //
+            // The calling function will think it owns the return mesh, so
+            // increment its reference count.
+            //
+            curves[i]->Register(NULL);
+            return curves[i];
+        }
+    }
+
+    EXCEPTION1(InvalidVariableException, name);
+}
+
+
+// ****************************************************************************
+//  Method: avtCurve3DFileFormat::GetVar
+//
+//  Purpose:
+//      Returns a variable.
+//
+//  Notes:      This is not meaningful for this file type and is here only to
+//              meet the base type's interface.
+//
+//  Arguments:
+//      <unused>  The domain number.
+//      name      The variable name.
+//
+//  Programmer:   Hank Childs
+//  Creation:     May 28, 2002
+//
+//  Modifications:
+//
+//    Hank Childs, Fri Aug  1 21:19:28 PDT 2003
+//    Retro-fit for STSD.
+//
+// ****************************************************************************
+
+vtkDataArray *
+avtCurve3DFileFormat::GetVar(const char *name)
+{
+    EXCEPTION1(InvalidVariableException, name);
+}
+
+
+// ****************************************************************************
+//  Method: avtCurve3DFileFormat::PopulateDatabaseMetaData
+//
+//  Purpose:
+//      Sets the database meta-data for this curve file.  There is only one
+//      mesh, the curve.  Each curve gets its own domain,for easy subselection.
+//
+//  Programmer: Hank Childs
+//  Creation:   May 28, 2002
+//
+//  Modifications:
+//
+//    Hank Childs, Tue Apr  8 11:09:03 PDT 2003
+//    Make sure we have read in the file first.
+//
+//    Hank Childs, Fri Aug  1 21:01:51 PDT 2003
+//    Mark curves as "curve" type.
+//
+//    Kathleen Bonnell, Thu Aug  3 08:42:33 PDT 2006 
+//    Added DataExtents to CurveMetaData. 
+//
+//    Kathleen Bonnell, Tue Jan 20 11:04:33 PST 2009
+//    Added SpatialExtents to CurveMetaData. 
+//
+//    Brad Whitlock, Mon Jul 23 12:07:07 PDT 2012
+//    Iterate based on curveNames since curves will not be populated on mdserver.
+//
+//    Olivier Cessenat, Thu Oct  5 18:41:09 CEST 2023
+//    One unstructured mesh per segment, then a gloval mesh with materials.
+//
+// ****************************************************************************
+
+void
+avtCurve3DFileFormat::PopulateDatabaseMetaData(avtDatabaseMetaData *md)
+{
+    if (!readFile)
```

Comment:
This method is executed on both the `mdserver` and the `engine`. However, only the `engine` needs the actual curve data.

If we encounter curves with a lot of points (say > 10^6) having to *read* the *whole* file *and* store its contents in memory is bad to do on the `mdserver`. It would be better to do this...

```
if (!readFile)
{
#ifdef MDSERVER 
    bool const clearData = true;
    ReadFile(clearData);
#else
    ReadFile();
#endif  
}
```

and then down in `ReadFile(bool clearData = false)` you would clear out the `vector<double>` arrays before returning. That way, you still suffer reading all the data and briefly storing it but the `mdserver` doesn't hang onto it after its finished what it needs to populate the meta data.

Reply:
Thanks for the tip.